core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -289,13 +302,21 @@ public synchronized InetSocketAddress getAddress() throws UnavailableException {
   protected synchronized <V> V retryRPC(RpcCallable<V> rpc) throws AlluxioStatusException {
     RetryPolicy retryPolicy = mRetryPolicySupplier.get();
     Exception ex = null;
+    Timer.Context metricContext = null;
+    if (rpc.getName() != null) {
+      metricContext = MetricsSystem.clientTimer(getQualifiedMetricName(rpc.getName())).time();
+    }
     while (retryPolicy.attempt()) {
       if (mClosed) {
         throw new FailedPreconditionException("Client is closed");
       }
       connect();
       try {
-        return rpc.call();
+        V result = rpc.call();
+        if (metricContext != null) {
+          metricContext.close();
```

Comment:
shouldn't this be in `finally`?

Reply:
The semantic is slightly different, but I think you're right we shouldn't leave the timer object not closed. It doesn't look like Codahale has support for canceling a timer after it has been started. I've added it to a try-finally - the side effect is that it will time failures in addition to successful RPCs.